akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -161,7 +164,26 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
     dir
   }
 
-  private final class SnapshotFilenameFilter(persistenceId: String) extends FilenameFilter {
+  private def extractMetadata(filename: String): Option[(String, Long, Long)] = {
+    val sequenceNumberEndIdx = filename.lastIndexOf('-')
+    val persistenceIdEndIdx = filename.lastIndexOf('-', sequenceNumberEndIdx - 1)
+    val timestampString = filename.substring(sequenceNumberEndIdx + 1)
+    if (persistenceIdStartIdx >= persistenceIdEndIdx || timestampString.exists(!_.isDigit)) None
+    else {
+      val persistenceId = filename.substring(persistenceIdStartIdx, persistenceIdEndIdx)
+      val sequenceNumber = filename.substring(persistenceIdEndIdx + 1, sequenceNumberEndIdx).toLong
+      val timestamp = filename.substring(sequenceNumberEndIdx + 1).toLong
```

Comment:
Addressed now, entire thing is wrapped in a try and we log a warning if unable to extract metadata.